ui/components/ui/identicon/identicon.component.js

```diff
@@ -84,13 +85,11 @@ export default class Identicon extends PureComponent {
       return this.renderImage();
     }
 
-    if (address) {
-      if (isHex(address)) {
-        const checksummedAddress = checksumAddress(address);
+    if (isHexString(addHexPrefix(address))) {
```

Comment:
Ok, down to one last issue! I think this is an excellent improvement for the old `if (isHex(address)) { ... }` block, but the `if (address)` part should be restored. Otherwise we're preventing jazzicons from being rendered for token/currency symbols (which, confusingly, are supplied using the `address` prop).

Reply:
```
    if (address) {
      const checksummedAddress =
        isHexString(addHexPrefix(address)) &&
        checksumAddress(addHexPrefix(address));

      if (checksummedAddress && contractMap[checksummedAddress]?.logo) {
        return this.renderJazzicon();
      }

      return (
        <div
          className={classnames({ 'identicon__address-wrapper': addBorder })}
        >
          {useBlockie ? this.renderBlockie() : this.renderJazzicon()}
        </div>
      );
    }
```
Think that does it!